nixos/modules/hardware/printers.nix

```diff
@@ -0,0 +1,111 @@
+  # TODO: is it possible to combine the forbiddenChars with that regex?
+  # test by using `nix repl` and then `builtins.match "[[:graph:]]+" "teststring#"`
+  printerName = (types.addCheck (types.strMatching "[[:graph:]]+") forbiddenChars)
+    // { description = "printable string without spaces, # and /"; };
+
+
+in
+{
+  options = {
+    hardware.printers = {
+      ensureDefaultPrinter = mkOption {
+        type = types.nullOr printerName;
+        default = null;
+        description = ''
+          Ensures the named printer is the default CUPS printer / printer queue.
+        '';
+      };
+      ensurePrinters = mkOption {
+        description = ''
+          Will regularly ensure that the given CUPS printers are configured as declared here.
+          If a printer's options are manually changed afterwards, they will be overwritten eventually.
+          This option will never delete any printer, even if removed from this list.
+          You can check existing printers with <literal>lpstat -s</literal>
+          and remove printers with <literal>lpadmin -x \<printer-name\></literal>.
+          Printers not listed here can still be manually configured.
+        '';
+        type = types.listOf (types.submodule {
+          options = {
+            name = mkOption {
+              type = printerName;
+              example = "BrotherHL_Workroom";
+              description = ''
+                Name of the printer / printer queue.
+                May contain any printable characters except "/", "#", and space.
+              '';
+            };
+            location = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "Workroom";
+              description = ''
+                Optional human-readable location.
+              '';
+            };
+            description = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              example = "Brother HL-5140";
+              description = ''
+                Optional human-readable description.
+              '';
+            };
+            deviceUri = mkOption {
+              type = types.str;
+              example = [
+                "ipp://printserver.local/printers/BrotherHL_Workroom"
+                "usb://HP/DESKJET%20940C?serial=CN16E6C364BH"
+              ];
+              description = ''
+                How to reach the printer.
+                <literal>lpinfo -v</literal> shows a list of supported device URIs and schemes.
+              '';
+            };
+            model = mkOption {
+              type = types.str;
+              example = "gutenprint.5.2://brother-hl-5140/expert";
+              description = ''
+                Location of the ppd driver file for the printer.
+                <literal>lpinfo -m</literal> shows a list of supported models.
+              '';
+            };
+          };
+        });
+        default = [];
```

Comment:
Wouldn't hurt readability to move this lonely `default` statement up to its corresponding `mkDefault` :)

Reply:
Makes sense, moved up, asserting you meant `mkOption`. ;)